streams/src/test/java/org/apache/kafka/streams/processor/internals/tasks/DefaultTaskManagerTest.java

```diff
@@ -94,6 +103,114 @@ public void shouldAssignTaskThatCanBeProcessed() {
         assertNull(taskManager.assignNextTask(taskExecutor));
     }
 
+    private class AwaitingRunnable implements Runnable {
+        private final CountDownLatch awaitDone = new CountDownLatch(1);
+        private final AtomicBoolean shutdownRequested = new AtomicBoolean(false);
+        @Override
+        public void run() {
+            while (!shutdownRequested.get()) {
+                try {
+                    taskManager.awaitProcessableTasks();
+                } catch (final InterruptedException ignored) {
+                }
+                awaitDone.countDown();
+            }
+        }
+
+        public void shutdown() {
+            shutdownRequested.set(true);
+            taskManager.signalProcessableTasks();
+        }
+    }
+
+    @Test
+    public void shouldBlockOnAwait() throws InterruptedException {
```

Comment:
Waiting for the whole verification timeout for a successful test result seems long to me.  
Does `shouldReturnFromAwaitOnInterruption()` not imply that the thread is waiting? Do we miss anything if we remove `shouldBlockOnAwait()`?

Reply:
I think it would be nice to keep it. But yeah, now I realize why I set this to 100. How about keeping it with 100? `shouldReturnFromAwaitOnInterruption` could pass if we always return immediately.